File: frontend/src/app/models/user-file.ts
import { ExtensionType } from "./extension-type";

export interface UserFile {
  id: string;
  name: string;
  path: string;
  size: number;
  type: ExtensionType;
  extension: string;
  isFolder: boolean;
  uploadedOn: Date;
  storageLink: string;
}

File: backend/functions/models/user_file.py
from datetime import datetime
from typing import TypedDict


class UserFile(TypedDict):
    """
    UserFiles are JSON objects stored as dictionaries.
    """

    id: str
    name: str
    path: str
    size: int
    extension: str
    isFolder: bool
    uploadedOn: datetime
    storageLink: str
